components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/model/graph/js/JsServletResponse.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.authentication.framework.config.model.graph.js;
+
+import org.wso2.carbon.identity.application.authentication.framework.util.FrameworkConstants;
+
+import javax.servlet.http.HttpServletResponse;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Javascript wrapper for Java level HttpServletResponse.
+ * This provides controlled access to HttpServletResponse object via provided javascript native syntax.
+ * e.g
+ * response.headers.["Set-Cookie"] = ['crsftoken=xxxxxssometokenxxxxx']
```

Comment:
We need to have better cookie add
e.g. 
var newCookie = {'name': 'myName'};
newCookie.path = '/foo/bar';
response.cookies.push(newCookie);